Silicon/CommonSocPkg/Include/Library/GpioLibrary.h

```diff
@@ -0,0 +1,338 @@
+/** @file
+  The platform GPIO library header.
+
+  Copyright (c) 2020, Intel Corporation. All rights reserved.<BR>
+  SPDX-License-Identifier: BSD-2-Clause-Patent

```

Comment:
I saw 3 gpio library header files in this patch:  Library/GpioLib.h,  Library/GpioLibrary.h and Library/GpioInfoLib.h
Do we have to provide another platform specific library for Library/GpioLibrary.h? what's the relation with GpioInfoLib?

Reply:
GpioLib.h ---> is for Platforms to call the main Gpio configuration routines.
GpioLibrary.h ---> CommonSoc\Silicon\GpioLib exposes certain functions so that platforms can use
GpioInfoLib.h ---> Platform specific implementation of API that CommonSoc\Silicon\GpioLib expects.
GpioInternalLib.h --->Prototypes for functions used within CommonSoc\Silicon\GpioLib

Reply:
Got it, we have 3 gpio library for different level. Do you think it would be more clear with below name?
GpioLib.h ---> common API for all the Platforms to call the main Gpio configuration routines.
GpioSocLib.h ---> CommonSoc\Silicon\GpioSocLib exposes certain functions so that platforms can use
GpioPlatformLib.h ---> Platform specific implementation of API

Reply:
Agree. Will change the names.